tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        aws.create_relationship(registered_entity, registered_metric)
+
+        timestamp = getUTCmillis()
+
+        registered_metric.values.put((timestamp, 10))
+
+        expected_output = {
+            "edge_system_name": registered_entity.ref_entity.name,
+            "metric_name": registered_metric.ref_entity.name,
+            "metric_data": [{
+                "value": 10,
+                "timestamp": timestamp
+            }],
+            "unit": "null"
+        }
+
+        # Getting the parent child relationship array from registered metric
+        formatted_data = aws._format_data(registered_metric)
+
+        formatted_json_data = json.loads(formatted_data)
+
+        # Check two dicts are equal or not
+        self.assertEqual(self.dict_equal(formatted_json_data, expected_output), True,
+                         "Check implementation of _format_data")
+
+    def test_implementation_format_data_without_enclose_metatadata(self):
+        """
+        Method to test the implementation of create_relationship method without enclosed meta_data option
+        of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=False)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        aws.create_relationship(registered_entity, registered_metric)
+
+        # Get current timestamp
+        timestamp = getUTCmillis()
+
+        registered_metric.values.put((timestamp, 10))
+
+        # Expected output without enclosed metadata
+        expected_output = {
+            "metric_name": registered_metric.ref_entity.name,
+            "metric_data": [{
+                "value": 10,
+                "timestamp": timestamp
+            }],
+            "unit": "null"
+        }
+
+        # Getting the parent child relationship array from registered metric
+        formatted_data = aws._format_data(registered_metric)
+
+        # Convert json string to dict for the comparision
+        formatted_json_data = json.loads(formatted_data)
+
+        # Check two dicts are equal or not
+        self.assertEqual(self.dict_equal(formatted_json_data, expected_output), True,
```

Comment:
Refer 1st answer and use ordered(a)==ordered(b) and check for assertTrue or assertFalse http://stackoverflow.com/questions/25851183/how-to-compare-two-json-objects-with-the-same-elements-in-a-different-order-equa